src/test/java/core/basesyntax/daotest/WriteToFileImplTest.java

```diff
@@ -0,0 +1,84 @@
+package core.basesyntax.daotest;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.WriteToFile;
+import core.basesyntax.dao.daoimpl.WriteToFileImpl;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import org.junit.jupiter.api.Test;
+
+public class WriteToFileImplTest {
+    public static final String EMPTY_FILE_PATH = "src/test/resources/";
+    public static final String FILE_PATH_OK = "src/test/resources/output.csv";
+    public static final String WRONG_FILE_PATH = "src/test/resouRRRces/blabla";
+    private WriteToFile writer;
+
+    @Test
+    void writeToFile_Ok() {
+        writer = new WriteToFileImpl();
+        writer.writeToFile(report(), FILE_PATH_OK);
+
+        File file = new File(FILE_PATH_OK);
+        String actual;
+        try {
+            actual = String.join(System.lineSeparator(),
+                    Files.readAllLines(file.toPath()));
+        } catch (IOException e) {
+            throw new RuntimeException("Can`t write to file. " + e);
+        }

```

Comment:
In test cases, it is generally recommended to let any exceptions propagate to the testing framework, which will handle the exceptions and report them appropriately. By removing the try-catch block, the testing framework can accurately capture and report any exceptions that occur during the test execution.

Therefore, I recommend removing the try-catch block from the test method and letting any IOException propagate naturally. The testing framework will handle and report the exception, providing more accurate information about the failure.

If necessary, you can use assertion methods specific to file handling or exception handling to verify the expected behavior in case of an exception.

Reply:
not fixed